src/pages/recipes/[recipeId].tsx

```diff
@@ -1,14 +1,147 @@
 import { Loader } from "@/shared/components/Loader";
 import { trpc } from "@/utils/trpc";
 import { useRouter } from "next/router";
+import Image from "next/image";
+import { RiShareForwardFill } from "react-icons/ri";
+import { SnackbarProvider } from "@/components/Snackbar";
 
 export default function Recipe() {
   const router = useRouter();
   const { data, isError, isLoading } = trpc.recipes.getRecipe.useQuery({
     recipeId: String(router.query.recipeId),
   });
+
   if (isLoading) {
     return <Loader />;
   }
-  return <div>Unfinished</div>;
+
+  return (
+    <div className="flex flex-col items-center pb-10">
+      <div className="w-2/3 space-y-6">
+        <div className="text-center text-4xl font-bold">{data!.name}</div>
+        <div>
+          {data!.description}
+          <br />
+          <br />
+          <div className="text-sm">
+            Recipe by <b>{data!.author.name}</b>
+          </div>
+        </div>
+        <button
+          onClick={() => copy()}
+          className="flex flex-row border-4 border-solid border-red-400 p-2 px-10 font-bold hover:bg-red-400 gap-3"
+        >
+          Share
+          <RiShareForwardFill size={22}/>
+        </button>
+        <div>
+          <Image
+            width={600}
+            height={600}
+            unoptimized
+            loading="lazy"
+            alt={data!.name}
+            src={data!.mainImage}
+          />
+        </div>
+        <div className="flex">
+          <div className="flex flex-col gap-6 bg-amber-100 p-5">
+            <div className="flex flex-row gap-6">
+              <div>
+                Preperation time: <br /> {+data!.prepTime!} minutes
```

Comment:
Converting with ```+``` can be a bit unclear sometimes. How about doing something like ```data.prepTime?.toNumber()```. This should work because if preptime is null, it wouldn't run toNumber() and will return null. Null is evaulated to 0 in jsx.

Reply:
Currently getting this error. It's the same error that I was talking about before.

<img width="959" alt="image" src="https://user-images.githubusercontent.com/99298978/234696371-e8a49e9b-171d-484a-b864-9292a2a73b0d.png">

Reply:
I think something is going on with the serialization/deserializion of the Decimal type. If you call toNumber() on the backend then it should be fine.

Reply:
I wasn't able to figure out how to use toNumber() in the backend (mainly because I didn't know how to convert the decimal to a number), so I ended up doing Number(data.prepTime) and Number(data.cookTime). It seems to work, but I'm not sure if it's conventional.

Reply:
Oh does data.prepTime.toNumber() not work?

Reply:
Yeah it wouldn't work on the frontend. It still said that it was a decimal.

Reply:
Can you do that in the backend?

Reply:
Yeah that's what I tried in the backend (In the recipes.ts page). I used it there and it wouldn't change the type of the prepTime and cookTime.